### DISCUSSION TITLE
Bug? eps input file BoundingBox not honored in output.

### DISCUSSION THREAD

**Discussion by alanngnet** [ImageMagick/ImageMagick#2632] [Help]
Two questions: 1) Did I discover a bug and 2) If so, how should I report it?

Context:
- Fresh install of imagemagick on MacOS 10.15.16 using homebrew. Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-0

Command line in Terminal:
`convert -density 600 test.eps  -colorspace Gray test.png`

Input test.eps and resulting output test.png are attached in the zip file here.
[Archive.zip](https://github.com/ImageMagick/ImageMagick/files/5286103/Archive.zip)

Problem: The boundingbox defined in test.eps is not respected. Compare whitespace difference when viewing test.eps vs. resulting test.png in Preview.

Expectation: I learned somewhere during my hours of research that ImageMagick does eps rendering by invoking GhostScript and supposedly the default when doing so is to include the GhostScript option `-dEPSCrop`. That is not happening? 

Evidence: My current workaround is to invoke ghostscript directly:
`gs -dSAFER -dEPSCrop -r600 -sDEVICE=pngalpha -o test.png test.eps`
The fact that Ghostscript is functioning nominally on my system might be evidence that the problem is a bug in how  ImageMagick is invoking Ghostscript?

Also odd is that doing all this with IM `convert` and the same input files was working fine for years for me until the moment I updated ImageMagick recently using HomeBrew. I don't know what my old version of ImageMagick (or Ghostscript) was before the upgrade, sorry.

(Explanation: I am actually doing a bunch of other ImageMagick operations on these images, so it would be much more efficient for me to  continue doing all of this in a single 'convert' command instead of first running it through GS to generate png and then feeding that output to 'convert' for the rest of the operations.)

**Comment by dlemstra** [comment]
It's not a bug. If you want to enable `-dEPSCrop` you will need to pass the option `-define eps:use-cropbox=true` before reading the image.

**Reply by alanngnet** [reply]
In that case my suspicion of an ImageMagick bug is even stronger, because I get the exact same result as above when I execute 
`convert -define eps:use-cropbox=true -density 600 test.eps -colorspace Gray test.png`

Attached are the input and output files from the above command: [Archive.zip](https://github.com/ImageMagick/ImageMagick/files/5286685/Archive.zip)

I had previously discovered and tried this flag based on the documentation here https://imagemagick.org/script/defines.php. I also noted that that page implies the default setting is 'true' meaning there should be no need to set the flag. This is also how ImageMagick worked for me in this process for many years until a few days ago when I updated it.

By the way, one of the other many failed attempts to get ImageMagick to cooperate was to add a PostScript CropBox definition in the EPS header *in addition* to the BoundingBox. Even then ImageMagick, with the use-cropbox flag set to true, gave the same uncropped result.

**Reply by alanngnet** [reply]
I believe my two questions here remain unanswered?

**Reply by alanngnet** [reply]
Also note that omitting the -define eps:use-cropbox=true flag makes no difference, since both (as expected and documented) include -dEPSCrop by default in their invocation of GS.  Evidence:
`convert -verbose -density 600 test.eps -colorspace Gray test.png`
results in
`/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-oy1_Z6IvuzB7DfikIMLU1tcSBbdvQ8zB1 PNG 5100x6600 5100x6600+0+0 8-bit sRGB 167038B 4.020u 0:00.762
test.eps EPS 5100x6600 5100x6600+0+0 16-bit sRGB 167038B 0.000u 0:00.000
[ghostscript library 9.52] -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r600x600' -g5100x6600 -dEPSCrop  '-sOutputFile=/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-oy1_Z6IvuzB7DfikIMLU1tcSBbdvQ8zB%d' '-f/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-ZpR7NblEl5sxi5pM9I5LuYAyWZp8d-Rk' '-f/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-LVX074YOidKv4A_bV5JbT5Wz3TC9dfSe'test.eps=>test.png EPS 5100x6600 5100x6600+0+0 16-bit Grayscale Gray 167038B 13.240u 0:01.732`

**Comment by snibgo** [comment]
I suggest you use "-verbose" in your command. This should show you the Ghostscript command that is generated. If you have "-define eps:use-cropbox=true" but there is no "-dEPSCrop" in the GS command, I suppose that is a bug.

There is an "issues" forum for bugs: https://github.com/ImageMagick/ImageMagick/issues

**Reply by alanngnet** [reply]
Thank you. The result of 
`convert -verbose -define eps:use-cropbox=true -density 600 test.eps -colorspace Gray test.png`

is:
`/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-qm8U1-FiM3pgs8-Cpy5AWjXPhWSe1FYh1 PNG 5100x6600 5100x6600+0+0 8-bit sRGB 167038B 3.990u 0:00.794
test.eps EPS 5100x6600 5100x6600+0+0 16-bit sRGB 167038B 0.000u 0:00.000
[ghostscript library 9.52] -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r600x600' -g5100x6600 -dEPSCrop  '-sOutputFile=/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-qm8U1-FiM3pgs8-Cpy5AWjXPhWSe1FYh%d' '-f/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-p6zA8gDQYUwXQ6ZqkP_teiyoKMiM5TbI' '-f/var/folders/hl/qyt_6pkx5sq5ggbk311zgvr80000gn/T/magick-MZZW8Qozta73N_PCZ5THlBzkl_LLZQgt'test.eps=>test.png EPS 5100x6600 5100x6600+0+0 16-bit Grayscale Gray 167038B 12.930u 0:01.743`

Again the resulting .png file is uncropped. We do see that -dEPSCrop is supposedly included in the GS command that IM issues. In my original post above we see that if I invoke GS directly things work fine. So is there some other problem with the way IM is invoking GS here?

**Reply by t246246** [reply]
If I understand this discussion correct, it looks gs honor -g5100x6600.  To get the desired result, IM should add -g2658x425, as bounding box size in your input file is 319x51 at default 72 dpi, so as to set it at 600 dpi.

**Comment by fmw42** [comment]
If I recall, in the past, identify -verbose used to show both a geometry and an image geometry. The second was what was need. The first for this image is Geometry: 612x792+0+0 which is too large. But I do not see listed in the verbose info now the second, image geometry

**Reply by rbherrmann** [reply]
I noticed a  problem with  version 7.1.1-8 downloaded using homebrew that was not in veSrion 6.9.11-60-5 downloaded using macports. 

If the EPS file had  an initial part
     %%BoundingBox: (atend)
and the BoundingBox at the end, the resulting image in png, e.g.,  convert test.eps test.png.  was cropped
If the correct BoundingBox is placed at the beginning of the file, the conversion is not cropped.

ImageMagick Version 6.9.1-60-5. does not have this problem
